mono/metadata/mono-hash.c

```diff
@@ -116,14 +93,51 @@ calc_prime (int x)
 }
 #endif
 
+#define HASH_TABLE_MAX_LOAD_FACTOR 0.5f
```

Comment:
0.5 was experimental (0.7 was leading to long chains). 0.05 random low value on the rationale that it can't be worse than not compacting at all. 3 is just for the sake of comparison, not to do more work than before (the previous code was using a second order linear recurrence that was more or less equivalent to this)